genesis/model/contract/smart_contract.go

```diff
@@ -76,7 +76,18 @@ func (sc *BaseSmartContract) CreateComposite(compositeFactory object.CompositeFa
 	if exist {
 		return nil, fmt.Errorf("delegate with name %s already exist", interfaceKey)
 	}
+
+	compositeAsChild, ok := composite.(object.Child)
```

Comment:
Why composite is not a child by default?

Reply:
Because we separate responsibility of interfaces. Composite about ability to be injected into smth, Child - about having a parent. If Composite also must have parent, than Composite interface must require GetParent() method specifically.